packages/indexer/native/src/lib.rs

```diff
@@ -110,6 +122,92 @@ impl NativeIndexer {
         }
         Ok(())
     }
+
+    fn publish_new_block_info(&self, block: &CoreBlockView) {
+        let emitters = self.emitters.read().unwrap();
+        let lock_script_emitters = emitters.iter().filter(|x| x.lock.is_some());
+        let type_script_emitters = emitters.iter().filter(|x| x.type_.is_some());
+
+        let transactions = block.transactions();
+        for (_tx_index, tx) in transactions.iter().enumerate() {
+            // TODO: cell_consumed event ?
+            for (output_index, output) in tx.outputs().into_iter().enumerate() {
+                let lock_script = output.lock();
+                for emitter in lock_script_emitters.clone() {
+                    if lock_script == emitter.lock.clone().unwrap() {
+                        if emitter.subscription_topic == 0 {
+                            self.emit_cell_generated_event(emitter, tx.hash(), output_index);
+                        } else {
+                            self.emit_transaction_generated_event(emitter, tx.hash());
+                        }
+                    }
+                }
+                if let Some(type_script) = output.type_().to_opt() {
+                    for emitter in type_script_emitters.clone() {
+                        if type_script == emitter.type_.clone().unwrap() {
+                            if emitter.subscription_topic == 0 {
+                                self.emit_cell_generated_event(emitter, tx.hash(), output_index);
+                            } else {
+                                self.emit_transaction_generated_event(emitter, tx.hash());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn publish_fork_info(&self, tip_number: u64, tip_hash: Byte32) {
```

Comment:
A rollback event is just the reverse of a normal block: you revert all generated cells, and commit all consumed cells. We should test against the cells in the block, and generate the same cell notification events.

Reply:
fixed.